lalrpop/src/normalize/token_check/test.rs

```diff
@@ -163,6 +163,27 @@ fn match_catch_all() {
     assert!(validate_grammar(&grammar).is_ok())
 }
 
+/// Test that a `catch-all` can be use in the first `match` arm.
+/// Before the pull request to close [issue 325](https://github.com/lalrpop/lalrpop/issues/325),
+/// the usage of the `catch-all` symbol was not allowed in the first arm of a `match` block.
+fn match_catch_all_in_first_arm() {
+    let grammar = r#"
+        grammar;
+        match {
+            _
+        } else {
+            r"\w+"
+        }
+        pub Term = {
+            Num,
+            "(" <Term> ")",
+            r"\w+" => format!("Id({})", <>),
+        };
+        Num: String = r"[0-9]+" => <>.to_string();
+"#;
+    assert!(validate_grammar(&grammar).is_ok())
```

Comment:
Great! 

I wonder, can we use `check_intern_token` to also test how things are classified? In particular, the precedence. If we add `r"[a-z]"` as above, then we would expect that a single letter (like `x`) would match `r"[a-z]"` (which has higher precedence), but multiple letters would match `r"\w+"`:

```rust
check_intern_token(
    grammar,
    vec![
        ("x", r##"Some(r#"[a-z]"#, "x")##),
        ("xy", r##"Some(r#"\w+"#, "xy")##),
    ],
);
```